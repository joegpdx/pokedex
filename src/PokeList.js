import React, { Component } from 'react';
import PokeItem from './PokeItem.js'

export default class PokeList extends Component {
    render() {
        const pokedeck = this.props.pokemons
            .map((pokecard, i) => <PokeItem key={i} pokemon={pokecard} />);
        console.log(pokedeck)
        return (
            <ul>{pokedeck}</ul>
        )
    }
}