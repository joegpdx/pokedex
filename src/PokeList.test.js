import React from 'react';
import renderer from 'react-test-renderer';
import PokeList from './PokeList.js';
import pokeData from './Data.js';

test('renders PokeItem.js correctly', () => {
  const tree = renderer
    .create(<PokeList pokemons={pokeData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});