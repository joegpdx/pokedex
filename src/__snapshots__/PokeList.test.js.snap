// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokeItem.js correctly 1`] = `
<ul>
  <li>
    <div>
      <p>
        Pokemon: 
        venusaur
      </p>
      <img
        alt=""
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
      />
      <p>
        Attack: 
        82
      </p>
      <p>
        defense: 
        83
      </p>
      <p>
        HP: 
        80
      </p>
    </div>
  </li>
  <li>
    <div>
      <p>
        Pokemon: 
        bulbasaur
      </p>
      <img
        alt=""
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
      />
      <p>
        Attack: 
        49
      </p>
      <p>
        defense: 
        49
      </p>
      <p>
        HP: 
        45
      </p>
    </div>
  </li>
  <li>
    <div>
      <p>
        Pokemon: 
        ivysaur
      </p>
      <img
        alt=""
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/002.png"
      />
      <p>
        Attack: 
        62
      </p>
      <p>
        defense: 
        63
      </p>
      <p>
        HP: 
        60
      </p>
    </div>
  </li>
  <li>
    <div>
      <p>
        Pokemon: 
        charizard-mega-y
      </p>
      <img
        alt=""
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/006_f3.png"
      />
      <p>
        Attack: 
        104
      </p>
      <p>
        defense: 
        78
      </p>
      <p>
        HP: 
        78
      </p>
    </div>
  </li>
</ul>
`;
